features/step_definitions/stepdefs.js

```diff
@@ -0,0 +1,72 @@
+const assert = require("assert");
+const { Given, When, Then } = require("@cucumber/cucumber");
+const axios = require("axios");
+const { checkUserExists } = require("../utils/apiUtils");
+
+Given("check that the user with id {int} exists", async function (id) {
+  const userExists = await checkUserExists(id);
+  assert.ok(userExists);
+});
+
+When("a list of users is requested", async function () {
+  const resp = await axios.get("https://reqres.in/api/users");
+  this.listOfUsers = resp;
+});
+
+When("a list of users on page {int} is requested", async function (page) {
+  const resp = await axios.get(`https://reqres.in/api/users?page=${page}`);
```

Comment:
Would be nice to extract all call to the server to a separate util client e.g.: `ReqresClient.js` and store all the rest logic there

```js
const axios = require('axios');

const apiClient = axios.create({
    baseURL: 'https://reqres.in/api',
    timeout: 1000,
});

const createUser = async (userData) => {
    try {
        const response = await apiClient.post('/users', userData);
        console.log(`User created with ID: ${response.data.id}`);
        return response.data;
    } catch (error) {
        console.error('Error creating user:', error);
        throw error;
    }
};

const getUser = async (userId) => {
    try {
        const response = await apiClient.get(`/users/${userId}`);
        console.log(`User fetched with ID: ${userId}`);
        return response.data;
    } catch (error) {
        console.error(`Error getting user with ID ${userId}:`, error);
        throw error;
    }
};

const updateUser = async (userId, userData) => {
    try {
        const response = await apiClient.put(`/users/${userId}`, userData);
        console.log(`User updated with ID: ${userId}`);
        return response.data;
    } catch (error) {
        console.error(`Error updating user with ID ${userId}:`, error);
        throw error;
    }
};

const deleteUser = async (userId) => {
    try {
        const response = await apiClient.delete(`/users/${userId}`);
        console.log(`User deleted with ID: ${userId}`);
        return response.status === 204;
    } catch (error) {
        console.error(`Error deleting user with ID ${userId}:`, error);
        throw error;
    }
};

module.exports = {
    createUser,
    getUser,
    updateUser,
    deleteUser,
};
```

then you can use it in your step definitions
```js
const { createUser, getUser, updateUser, deleteUser } = require('./apiClient');
```

Please consider reading [SLAP](https://medium.com/@outfunnel/the-single-level-of-abstraction-principle-52b5d56ef54b)
- TLDR; To ensure clarity and maintain a consistent level of abstraction, each method should encapsulate a single level of detail. This mirrors real-life conversations where we maintain a consistent detail level. For instance, when discussing my weekend, I would mention highlights like visiting a city or enjoying a nice dinner, but not minute actions like locking the door when I left my apartment. This approach keeps the code clean and focused.